### PR TITLE
feat: add HardwareDetector interface and measurement keys for NFD integration

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
## Summary

Task 0 of NFD Snapshot Enrichment (Track A).

Adds shared infrastructure for day-0 GPU hardware detection:
- Measurement keys: `gpu-present`, `driver-loaded`, `detection-source` (with `KeyGPU*` prefix convention)
- Timeout constant: `NFDDetectionTimeout` (5s)
- `HardwareDetector` interface and `HardwareInfo` type (including `DetectionSource` field)

## Note
The NFD Go dependency (`sigs.k8s.io/node-feature-discovery`) and PCI/NFD constants
(vendor ID, device classes, source names) are deferred to Task 1 when code imports
NFD packages. `go mod tidy` removes unused imports, and `golangci-lint` flags
unused constants, so both are best added alongside their first consumer.

## Testing
- `go build ./...` passes
- All existing tests pass with `-race`
- Error-path test case validates nil-safety of HardwareDetector mock

### CODE REVIEW

File: pkg/collector/gpu/hardware_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			}
+			if info.GPUCount != tt.wantCount {
+				t.Errorf("GPUCount = %v, want %v", info.GPUCount, tt.wantCount)
+			}
+			if info.DriverLoaded != tt.wantDriver {
+				t.Errorf("DriverLoaded = %v, want %v", info.DriverLoaded, tt.wantDriver)
+			}
+		})
+	}
+}
```
Comment (mchmarny) [inline pkg/collector/gpu/hardware_test.go]: **Important: Missing error-path test case.**

The table has `wantErr` wired up but no test case actually exercises `wantErr: true`. An error case is important because it validates the interface contract (callers must handle errors from `Detect`). Please add a case like:

```go
{
    name: "detection failure",
    detector: &mockHardwareDetector{
        info: nil,
        err:  errors.New(errors.ErrCodeInternal, "sysfs not available"),
    },
    wantErr: true,
},
```

Also note: when `wantErr` is true and `info` is nil, the subsequent field checks (`info.GPUPresent`, etc.) will panic with a nil pointer dereference. The `return` after the error check only fires when the error expectation is wrong. You need an early `return` (or `continue`) when `err != nil && tt.wantErr`:

```go
if (err != nil) != tt.wantErr {
    t.Errorf("Detect() error = %v, wantErr %v", err, tt.wantErr)
    return
}
if err != nil {
    return // error expected and received, skip field assertions
}
```

**Review by mchmarny — CHANGES_REQUESTED** [review]
Thank you for this contribution -- solid start on the NFD integration infrastructure. The interface design is clean, the timeout constant is well-placed in `pkg/defaults`, and the measurement keys follow existing patterns. Tests are table-driven as required. Appreciated the thorough doc comments throughout.

**Summary of findings (see inline comments for details):**

**Critical (1):**
- The blank import of `sigs.k8s.io/node-feature-discovery/api/nfd/v1alpha1` in production code executes NFD's `init()` functions (K8s scheme registration) as a side-effect in every binary. This dependency should either be deferred to Task 1 when it is actually used, or isolated behind a `//go:build tools` tag.

**Important (2):**
- `HardwareInfo` is missing a `DetectionSource` field despite `KeyDetectionSource` being added as a measurement key. The struct should be self-describing.
- The test is missing an error-path case (`wantErr: true`), and the current test structure would panic on nil `info` if one were added. Needs a nil guard after the error check.

**Minor (2):**
- `KeyDriverLoaded` and `KeyDetectionSource` break the `KeyGPU*` naming convention used by neighboring constants. Consider `KeyGPUDriverLoaded` and `KeyGPUDetectionSource`.
- The timeout constant placement and value are fine; just confirming the 5s headroom is intentional.

**PR description vs. actual diff:**
The description mentions "PCI/NFD constants: NVIDIA vendor ID, GPU device classes" but no such constants (e.g., `0x10de` vendor ID, `0x0300`/`0x0302` device classes) appear in the diff. Either the description should be updated to match the actual scope, or those constants should be added if they were intended for this PR.

**Review by mchmarny — CHANGES_REQUESTED** [review]
Thanks for the contribution — the interface design is clean and the table-driven test structure follows project conventions well.

A few things to address before merging:

1. **Blank NFD import** (`hardware.go:20-22`) — runs `init()` side-effects in all binaries for an unused dependency. Remove it or gate it behind a `//go:build tools` tag.
2. **Missing `DetectionSource` field** (`hardware.go:36-44`) — `KeyDetectionSource` exists as a measurement key but `HardwareInfo` has no matching field.
3. **Test error path** (`hardware_test.go:72-78`) — no `wantErr: true` case, and the assertions would panic on `nil` info if one were added.
4. **Naming consistency** (`types.go:37-39`) — `KeyDriverLoaded`/`KeyDetectionSource` break the `KeyGPU*` prefix pattern.

Also, the PR description mentions "PCI/NFD constants: NVIDIA vendor ID, GPU device classes" but these don't appear in the diff — might want to update the description or defer that bullet to the relevant task.

See inline comments for details and suggestions.

**Review by mchmarny — APPROVED** [review]
thanks for the fixes

/lgtm
